docs/grading.md

```diff
@@ -8,94 +8,121 @@ contributions and community involvement into appropriate grades. Generally
-| 2            | 3          | 3 absences, 2 bonus sessions   | 3                        | 2                         | 1                         | X                     | X                    |
-| 3            | 4          | 2 absences, 2 bonus sessions   | 5                        | 2                         | 1                         | X                     | Yes                  |
-| 4            | 4          | 2 absences, 2 bonus sessions   | 7                        | 2                         | 1                         | Yes                   | Yes                  |
+The proposal and features therein must be approved by a mentor at the start of the semester.
+However, it is important to remember that the proposal is a living document.
+Plans often change, and that is OK. In the event you decide to change your plans for the semester,
+or fail to adhere to the timeline set in the proposal, the proposal must be revised
+accordingly and the revisions must be approved by a mentor.
 
-## Grading Metrics
+### Features
+
+A feature is functional increment of your project. For software projects, a feature should
+typically be a working collection of commits that accomplished a portion of your overall goals.
+
+It is strongly recommended that features be broken down into sub tasks, which should be tracked
+using Github Issues. Issues should have well defined acceptance criteria, and should be organized
+using Github Projects. 
+
+A feature does not necessarily have to a functional change to your project if other work is necessary.
+Examples of non-functional features include learning or research goals, significant architectural or
+design work, etc.
+
+If a feature is not a functional change, there should be a greater focus on documentation
+to appropriately convey the work that was done.
+
+Progress for each feature must be meticulously documented in one or more blog posts. See
+[Blog Posts](#blog-posts) for more information.
+
+If you are not sure how to best divide up your project into features, please ask a mentor for assistance.
+
+## Documentation
+
+Projects should be well documented. Documentation can take many forms. A README and an
+OSI approved license are mandatory. The README should include a summary of the project,
+instructions for contributing, and any other information needed to develop for and use
+the project.
+
+Additional documentation (other than a README and blog posts) is required.
+Examples of additional documentation include but are not limited to UML diagrams,
+API documents, user manuals, FAQs, setup guides, troubleshooting information, etc.
+
+The use of a wiki is strongly encouraged. Creating a static project site can help you
+organize your various forms of documentation and provide excellent exposure for your
+project.
+
+If you are unsure how to properly document your project, please consult your mentor for
+advice early on.
 
```

Comment:
Documentation requirements are separate from features. I think all group members should be equally responsible for documentation.